src/refactor_nrepl/externs.clj

```diff
@@ -0,0 +1,28 @@
+(ns refactor-nrepl.externs
+  (:import java.net.URL
+           org.projectodd.shimdandy.ClojureRuntimeShim
+           java.net.URLClassLoader
+           java.util.ArrayList))
+
+(def rt (atom nil))
+
+(defn create-runtime []
+  (let [urls [(URL. "file:libs/shimdandy-impl-1.1.0.jar")
+              (URL. "file:libs/clojure-1.6.0.jar")
+              (URL. "file:libs/pomegranate-0.3.1-SNAPSHOT.jar")]
+        cl (URLClassLoader. (into-array URL urls))]
+    (reset! rt (ClojureRuntimeShim/newRuntime cl "refactor-nrepl"))))
+
+(defn add-dependencies [dep repos]
+  (when-not @rt
+    (create-runtime))
+  (.require @rt (into-array String ["cemerick.pomegranate"]))
```

Comment:
@tobias I was testing this at the repl (hoping to solve the jar-in-jar problem later, @benedekfazekas  suggested just unpacking everything into a dir in the jar and pointing the classloader in that direction) but I still couldn't get the pomegranate code to eval properly. 

I think it finds the lib folder and the jars correctly.  Since the shim runtime is created it must find, `shimdandy-impl` right?  

Inheriting from existing classloaders is OK, I think.  If I understand how it works then the nearest classloader is searched first and then if something isn't found the parent is searched.  As long as we're not polluting the user's classpath or classloader we should be golden.

Thanks for taking a look!